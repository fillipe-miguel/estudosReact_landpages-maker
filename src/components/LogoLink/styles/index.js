import styled, { css } from 'styled-components';

export const LogoLinkRoot = styled.a`
  ${() => css`
    display: flex;
    justify-content: center;
    max-height: 8rem;
    width: 20rem;
  `}
`;

export default LogoLinkRoot;
