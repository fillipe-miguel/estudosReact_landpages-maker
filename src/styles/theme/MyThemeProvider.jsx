import P from 'prop-types';
import React from 'react';
// Styles
import { ThemeProvider } from 'styled-components';
import GlobalStyles from '../global-styles';
import theme from './theme';

function MyThemeProvider({ children }) {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
}

MyThemeProvider.propTypes = {
  children: P.node.isRequired,
};

export default MyThemeProvider;
