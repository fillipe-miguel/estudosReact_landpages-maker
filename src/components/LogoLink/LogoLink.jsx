import P from 'prop-types';
import React from 'react';

import * as Styled from './styles';

import Heading from '../Heading';

function LogoLink({ text, srcImage, link }) {
  return (
    <Styled.LogoLinkRoot href={link}>
      {srcImage ? (
        <img src={srcImage} alt={text} />
      ) : (
        <Heading uppercase level="2" size="medium">
          {text}
        </Heading>
      )}
    </Styled.LogoLinkRoot>
  );
}

LogoLink.defaultProps = {
  srcImage: '',
};

LogoLink.propTypes = {
  text: P.string.isRequired,
  link: P.string.isRequired,
  srcImage: P.string,
};

export default LogoLink;
