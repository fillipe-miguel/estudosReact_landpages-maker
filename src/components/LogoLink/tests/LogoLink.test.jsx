/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';
// Styles
import MyThemeProvider from '../../../styles/theme/MyThemeProvider';
// Components
import LogoLink from '../LogoLink';

describe('<LogoLink/>', () => {
  // ================================= Teste 01 =================================
  it('should render without image', () => {
    render(
      <MyThemeProvider>
        <LogoLink text="Meu Título" link="#" />
      </MyThemeProvider>,
    );

    expect(
      screen.queryByRole('heading', { name: 'Meu Título' }),
    ).toBeInTheDocument();
  });

  // ================================= Teste 02 =================================
  it('should render with image', () => {
    render(
      <MyThemeProvider>
        <LogoLink
          text="Meu Título com imagem"
          link="#"
          srcImage="public\assets\images\logo.svg"
        />
      </MyThemeProvider>,
    );

    expect(
      screen.queryByRole('img', { name: 'Meu Título com imagem' }),
    ).toBeInTheDocument();
  });

  // ================================= Snapshot =================================
  it('should match snapshot', () => {
    const { container } = render(
      <MyThemeProvider>
        <LogoLink text="Meu Título com imagem" link="#" />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
