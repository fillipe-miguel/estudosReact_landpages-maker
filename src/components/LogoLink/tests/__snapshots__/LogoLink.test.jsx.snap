// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink/> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-height: 8rem;
  width: 20rem;
}

.c1 {
  color: #0A1128;
  text-transform: uppercase;
  margin-bottom: 3.2rem;
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c1 {
    font-size: 2.4rem;
  }
}

<a
  class="c0"
  href="#"
>
  <h2
    class="c1"
  >
    Meu TÃ­tulo com imagem
  </h2>
</a>
`;
